sql/stats: TestAtMostOneRunningCreateStats failed

sql/stats.TestAtMostOneRunningCreateStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9585963?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9585963?buildTab=artifacts#/) on release-23.1.0 @ [aaef49cc37bd71e0f108601499e8cdb7649f2159](https://github.com/cockroachdb/cockroach/commits/aaef49cc37bd71e0f108601499e8cdb7649f2159):

Fatal error:

```
panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead

```
Stack: 

```
goroutine 21632692 [running]:
testing.tRunner.func1.2({0x42692e0, 0x5cd9320})
	GOROOT/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x39f
panic({0x42692e0, 0x5cd9320})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:118 +0x2a9
panic({0x42692e0, 0x5cd9320})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x0?, {0x5d1b038, 0xc000082058})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x42692e0, 0x5cd9320})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc018bcc700, {0x5d1b038, 0xc000082058})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:507 +0x21d
panic({0x42692e0, 0x5cd9320})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/stats_test.TestAtMostOneRunningCreateStats.func2()
	github.com/cockroachdb/cockroach/pkg/sql/stats_test/pkg/sql/stats/create_stats_job_test.go:179 +0x1e8
github.com/cockroachdb/cockroach/pkg/sql/stats_test.TestAtMostOneRunningCreateStats(0xc00bb20000)
	github.com/cockroachdb/cockroach/pkg/sql/stats_test/pkg/sql/stats/create_stats_job_test.go:184 +0x6fe
testing.tRunner(0xc00bb20000, 0x4d742e0)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAtMostOneRunningCreateStats
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a29a19c212b7b40401087e2c497e19d3/logTestAtMostOneRunningCreateStats25693069
    test_log_scope.go:79: use -show-logs to present logs inline
    stopper.go:229: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /artifacts/tmp/_tmp/a29a19c212b7b40401087e2c497e19d3/logTestAtMostOneRunningCreateStats25693069
    stopper.go:229: panic: CREATE STATISTICS job which was expected to fail, timed out instead
--- FAIL: TestAtMostOneRunningCreateStats (14.18s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAtMostOneRunningCreateStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26966

sql/stats.TestAtMostOneRunningCreateStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9729154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9729154?buildTab=artifacts#/) on release-23.1.0 @ [df033322c25c433c0296029197ae6f7e9b5b7532](https://github.com/cockroachdb/cockroach/commits/df033322c25c433c0296029197ae6f7e9b5b7532):

Fatal error:

```
panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead [recovered]
	panic: CREATE STATISTICS job which was expected to fail, timed out instead

```
Stack: 

```
goroutine 24816863 [running]:
testing.tRunner.func1.2({0x4272c00, 0x5ce94d0})
	GOROOT/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x39f
panic({0x4272c00, 0x5ce94d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:118 +0x2a9
panic({0x4272c00, 0x5ce94d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x42ea8c?, {0x5d2b378, 0xc000082058})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x4272c00, 0x5ce94d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc0083d8680, {0x5d2b378, 0xc000082058})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:507 +0x21d
panic({0x4272c00, 0x5ce94d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/sql/stats_test.TestAtMostOneRunningCreateStats.func2()
	github.com/cockroachdb/cockroach/pkg/sql/stats_test/pkg/sql/stats/create_stats_job_test.go:179 +0x1e8
github.com/cockroachdb/cockroach/pkg/sql/stats_test.TestAtMostOneRunningCreateStats(0xc004e436c0)
	github.com/cockroachdb/cockroach/pkg/sql/stats_test/pkg/sql/stats/create_stats_job_test.go:184 +0x6fe
testing.tRunner(0xc004e436c0, 0x4d83340)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAtMostOneRunningCreateStats
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a29a19c212b7b40401087e2c497e19d3/logTestAtMostOneRunningCreateStats2252726524
    test_log_scope.go:79: use -show-logs to present logs inline
    stopper.go:229: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /artifacts/tmp/_tmp/a29a19c212b7b40401087e2c497e19d3/logTestAtMostOneRunningCreateStats2252726524
    stopper.go:229: panic: CREATE STATISTICS job which was expected to fail, timed out instead
--- FAIL: TestAtMostOneRunningCreateStats (15.94s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102006 sql/stats: TestAtMostOneRunningCreateStats failed [C-test-failure O-robot T-sql-queries branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAtMostOneRunningCreateStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -164,19 +164,10 @@ func TestAtMostOneRunningCreateStats(t *testing.T) {
 	}
 
 	autoStatsRunShouldFail := func() {
-		expectErrCh := make(chan error, 1)
-		go func() {
-			_, err := conn.Exec(`CREATE STATISTICS __auto__ FROM d.t`)
-			expectErrCh <- err
-		}()
-		select {
-		case err := <-expectErrCh:
-			expected := "another CREATE STATISTICS job is already running"
-			if !testutils.IsError(err, expected) {
-				t.Fatalf("expected '%s' error, but got %v", expected, err)
-			}
-		case <-time.After(time.Second):
-			panic("CREATE STATISTICS job which was expected to fail, timed out instead")
+		_, err := conn.Exec(`CREATE STATISTICS __auto__ FROM d.t`)
+		expected := "another CREATE STATISTICS job is already running"
+		if !testutils.IsError(err, expected) {
+			t.Fatalf("expected '%s' error, but got %v", expected, err)
 		}
 	}
 

